operator/internal/manifests/ingester.go

```diff
@@ -288,12 +294,11 @@ func configureIngesterGRPCServicePKI(sts *appsv1.StatefulSet, opts Options) erro
 
 // newIngesterPodDisruptionBudget returns a PodDisruptionBudget for the LokiStack
 // Ingester pods.
-func newIngesterPodDisruptionBudget(opts Options) *policyv1.PodDisruptionBudget {
+func newIngesterPodDisruptionBudget(opts Options) (*policyv1.PodDisruptionBudget, error) {
 	l := ComponentLabels(LabelIngesterComponent, opts.Name)
-	// Default to 1 if not defined in ResourceRequirementsTable for a given size
-	mu := intstr.FromInt(1)
-	if opts.ResourceRequirements.Ingester.PDBMinAvailable > 0 {
-		mu = intstr.FromInt(opts.ResourceRequirements.Ingester.PDBMinAvailable)
+	pdbMinAvailable := intstr.FromInt32(1)
+	if opts.Stack.Size == lokiv1.SizeOneXPico || opts.Stack.Size == lokiv1.SizeOneXMedium {
+		pdbMinAvailable = intstr.FromInt32(opts.Stack.Replication.Factor)
```

Comment:
IIRC here we will only want to 
```suggestion
	pdbMinAvailable := intstr.FromInt32(1)
	if opts.Stack.Size == lokiv1.SizeOneXDemo {
		pdbMinAvailable = intstr.FromInt(opts.ResourceRequirements.Ingester.PDBMinAvailable)
```
And we would instead set it in the same code function where we preform the validation